### PR TITLE
Goodfellow2013maxout_1+Zayne-sprague

### PR DESCRIPTION
**Pull request by Zayne-sprague**
First Draft

### CODE REVIEW

File: _summaries/goodfellow2013maxout_1.md [hunk 1]
```diff
@@ -1,19 +1,92 @@
...
+*TODO - how is this different than max-pooling for CNNs?  Adding this would be helpful to the review.
+*TODO - add the visuals where marked
+*TODO - add equations where marked
+*TODO - break up sections a bit neater.
+
+
+**_Main Idea_**
+
+Taking the element-wise max of an input across multiple weight matrices produces a non-linear activation function
+that can learn any convex function and take advantage of dropout layers while improving accuracy and allowing deeper networks.
```
Comment (jaywhang) [inline _summaries/goodfellow2013maxout_1.md:19]: I like this concise summary.  It describes exactly what maxout does, and provides a high-level intuition for why it works (i.e. approximating convex function as a combination of piecewise linear functions learned by each weight matrix)

File: _summaries/goodfellow2013maxout_1.md [hunk 2]
```diff
@@ -1,19 +1,92 @@
...
+
+**_How is it realized_**
+
+Maxout networks use "Maxout Layers".  
+
+Maxout Layers have multiple weight matrices and biases for the same input!  
+(tip: Think of convolutional feature maps, in MLPs or linear layers they are just extra weight matrices and biases)
+
+Maxout Layers takes the values from all the weight matrices in its layer, does the element wise multiplication with the
```
Comment (jaywhang) [inline _summaries/goodfellow2013maxout_1.md:28]: I think it would be help to include the equation that defines the maxout function.  For example,  $abc$

File: _summaries/goodfellow2013maxout_1.md [hunk 3]
```diff
@@ -1,19 +1,92 @@
...
+input per weight matrix, and then does an element wise maximum across all the different activations for that input.
+
+
+TODO* {{A cool visual should go here}}
+
+
+**_What does this mean_**
+
+Because each input to the layer has a set of weights that are going to be Maxxed over, 
+the activation function has "learnable weights".
```
Comment (jaywhang) [inline _summaries/goodfellow2013maxout_1.md:38]: Emphasis on maxout being a parametrized activation function with learnable weights is nice!

File: _summaries/goodfellow2013maxout_1.md [hunk 4]
```diff
@@ -1,19 +1,92 @@
...
+
+**_How is it realized_**
+
+Maxout networks use "Maxout Layers".  
+
+Maxout Layers have multiple weight matrices and biases for the same input!  
+(tip: Think of convolutional feature maps, in MLPs or linear layers they are just extra weight matrices and biases)
+
+Maxout Layers takes the values from all the weight matrices in its layer, does the element wise multiplication with the
+input per weight matrix, and then does an element wise maximum across all the different activations for that input.
```
Comment (jaywhang) [inline _summaries/goodfellow2013maxout_1.md:29]: I think it would be help to include the equation that defines the maxout function.   In particular, we could include (perhaps a paraphrased version of) the equation at the top of page 2 column 2.  The current description makes sense to me, but would be confusing if I had not read the paper beforehand.

File: _summaries/goodfellow2013maxout_1.md (hunk 1 above)
Comment (jaywhang) [inline _summaries/goodfellow2013maxout_1.md:19]: I like this concise summary.  It describes exactly what maxout does, and provides a high-level intuition for why it works (i.e. approximating convex function as a max (which itself is a convex function) of piecewise linear functions learned by each weight matrix).

One minor point I have is that perhaps we can be a bit more descriptive about the expressivity of maxout.  It can _approximate_ any convex functions, but it certainly can't learn "any convex function", especially in practical setting where the number of weight matrices is limited.

File: _summaries/goodfellow2013maxout_1.md [hunk 5]
```diff
@@ -1,19 +1,92 @@
...
+
+They achieve state-of-the-art results in some cases reducing error by as much as 4%.
+
+However, the biggest performance improvement is best shown by how much of the model is being utilized.  
+
+Other models that use rectifiers suffered from dead neurons which made the gradient suffer and learning stall,
+Maxout overcomes these issues.
+
+Because of this, Maxout is able to support deeper models and make better use of the parameters, indicating that 
+even better performance may be achievable.
```
Comment (jaywhang) [inline _summaries/goodfellow2013maxout_1.md:75]: Perhaps we can say "...and make _a more efficient_ use of the parameters", since parameter efficiency is less vague than "better use" of parameters.

File: _summaries/goodfellow2013maxout_1.md [hunk 6]
```diff
@@ -1,19 +1,92 @@
...
+
+TODO* {{graphs on the activations switching between positive and negatives should go here}}
+
+
+**_What are the costs_**
+
+The major cost is that you now have many more weights per layer than you would if you used a rectifier.
+
+
+**_How well does the paper perform?_**
```
Comment (jaywhang) [inline _summaries/goodfellow2013maxout_1.md:63]: It might be worth mentioning the universal approximability of maxout here, since a good chunk of the paper is spent on the theorems.  I think the intuition of the proof is pretty simple as well -- using maxout to represent piecewise linear functions with arbitrarily many "segments" and invoking Proposition 4.2

File: _summaries/goodfellow2013maxout_1.md [hunk 7]
```diff
@@ -1,19 +1,92 @@
...
+
+
 
 ## TL;DR
-* Three
-* Bullets
-* To highlight the core concepts
+- Maxout requires a single layer to have multiple (k) weight matrices and takes the maximum values of those k activations as the final output
+- Maxout Activations take advantage of Dropout more than any other activation function (at the time of the writing of the paper)
+- Maxout Layers help use the entire model whereas rectifiers (or other activations) can "turn off" activations by being set to 0
```
Comment (jaywhang) [inline _summaries/goodfellow2013maxout_1.md:92]: Helping "use the entire model" seems a bit ambiguous.  How about something like "Maxout alleviates the dead neuron issues prevalent with rectifiers by _learning_ a representation that is less sparse."?

File: _summaries/goodfellow2013maxout_1.md [hunk 8]
```diff
@@ -1,19 +1,92 @@
...
+Because of this, Maxout is able to support deeper models and make better use of the parameters, indicating that 
+even better performance may be achievable.
+
+
+**_Interesting Variants_**
+
+They show how their model fairs with MLPs (linear layers) and CNNs.  
+
+Beyond that, they mostly stick with the mathematical ramifications of why Maxout works, specifically on why it works best
+with dropout.
```
Comment (jaywhang) [inline _summaries/goodfellow2013maxout_1.md:83]: It might be helpful to discuss here some of the details on the effect of maxout activation on the optimization performance (Sec. 8) and how it works with dropout (Sec. 7).  While the arguments presented are not hard facts, they seem to carry interesting intuitions (e.g. maxout trained with dropout leads to each subnetwork learning to "specialize" in one thing, and maxout having better gradient flow compared to ReLU)

File: _summaries/goodfellow2013maxout_1.md (hunk 4 above)
Comment (Zayne-sprague) [reply _summaries/goodfellow2013maxout_1.md:29]: Very good call, adding them now.

File: _summaries/goodfellow2013maxout_1.md (hunk 1 above)
Comment (Zayne-sprague) [reply _summaries/goodfellow2013maxout_1.md:19]: Good eye, adding that clarification as "learn approximations of any convex function".

File: _summaries/goodfellow2013maxout_1.md (hunk 7 above)
Comment (Zayne-sprague) [reply _summaries/goodfellow2013maxout_1.md:92]: Oh, that's perfect! Thank you!!

File: _summaries/goodfellow2013maxout_1.md (hunk 8 above)
Comment (Zayne-sprague) [reply _summaries/goodfellow2013maxout_1.md]: I tried to add this, let me know what you think.

**Comment by Zayne-sprague** [comment]
Final Draft submitted

**Comment by Zayne-sprague** [comment]
Final Draft resubmitted with a super small fix.

**Review by zhaoyue-zephyrus — APPROVED** [review]
